### PR TITLE
Added Recipe Search Overlay

### PR DESCRIPTION
**Pull request by Conutik**
Added recipe search overlay which uses the same way ah and bz search overlays are made.

Known bugs that I couldn't fix:
- [ ] Ghost shadow appearing in different pages other than crafting menu without the pickaxe.
- [ ] Shift clicking an item crashes the game

(Third time's the charm)

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/*
+	/*
```
Comment (jani270) [inline build.gradle.kts:1]: What happend here lol

**Review by jani270 — CHANGES_REQUESTED** [review]
Please format your files with the Intellij built in formatter

File: build.gradle.kts (hunk 1 above)
Comment (Conutik) [reply build.gradle.kts:1]: I think I pressed tab by accident D:

**Comment by nea89o** [comment]
Hey, it looks like you accidentally build your PR based on https://github.com/Moulberry/NotEnoughUpdates instead of https://github.com/NotEnoughUpdates/NotEnoughUpdates. You might need to rebase some of your code on the latest commit from here.

**Comment by Conutik** [comment]
Fixed all comments above

**Comment by jani270** [comment]
> Fixed all comments above

Unfortunatly that is still wrong, see merge conflicts. Your branch is still based on Moulberry/NEU instead of NEU/NEU. Which makes this impossible to merge.

File: src/main/java/io/github/moulberry/notenoughupdates/options/NEUConfig.java [hunk 2]
```diff
@@ -377,6 +377,13 @@ public void executeRunnable(int runnableId) {
 	)
 	public BazaarTweaks bazaarTweaks = new BazaarTweaks();
 
+	@Expose
+	@Category(
+		name = "Recipe Tweaks",
+		desc = "Tweaks for the Recipe Search"
+	)
+	public BazaarTweaks recipeTweaks = new BazaarTweaks();
```
Comment (jani270) [inline src/main/java/io/github/moulberry/notenoughupdates/options/NEUConfig.java:385]: BazaarTweaks

File: src/main/java/io/github/moulberry/notenoughupdates/options/seperateSections/RecipeTweaks.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+		desc = "Show past searches below the autocomplete box"
+	)
+	@ConfigEditorBoolean
+	@ConfigAccordionId(id = 0)
+	public boolean showPastSearches = true;
+
+	@Expose
+	@ConfigOption(
+		name = "ESC to Full Close",
+		desc = "Make pressing ESCAPE close the search GUI without opening up the Bazaar again\n" +
```
Comment (jani270) [inline src/main/java/io/github/moulberry/notenoughupdates/options/seperateSections/RecipeTweaks.java:67]: Bazaar

File: src/main/java/io/github/moulberry/notenoughupdates/overlays/RecipeSearchOverlay.java [hunk 4]
```diff
@@ -0,0 +1,494 @@
...
+			desc.removeAll(title);
+
+			if (thisSearchId != searchId.get()) return;
+
+			Set<String> bazaarItems = NotEnoughUpdates.INSTANCE.manager.auctionManager.getBazaarKeySet();
+			// Amalgamated Crimsonite (Old) // TODO remove from repo
+			bazaarItems.remove("AMALGAMATED_CRIMSONITE");
+
+			title.retainAll(bazaarItems);
+			desc.retainAll(bazaarItems);
```
Comment (jani270) [inline src/main/java/io/github/moulberry/notenoughupdates/overlays/RecipeSearchOverlay.java:338]: Bazaar

File: src/main/java/io/github/moulberry/notenoughupdates/overlays/RecipeSearchOverlay.java [hunk 5]
```diff
@@ -0,0 +1,494 @@
...
+						}
+					} else if (mouseX < width / 2 + 75) {
+						searchStringExtra = "";
+						close();
+					} else if (mouseX < width / 2 + 100) {
+						searchStringExtra = "";
+						close();
+						Minecraft.getMinecraft().thePlayer.sendQueue.addToSendQueue(new C0DPacketCloseWindow(Minecraft.getMinecraft().thePlayer.openContainer.windowId));
+						NotEnoughUpdates.INSTANCE.openGui =
+							new GuiScreenElementWrapper(new NEUConfigEditor(NotEnoughUpdates.INSTANCE.config, "Bazaar Tweaks"));
```
Comment (jani270) [inline src/main/java/io/github/moulberry/notenoughupdates/overlays/RecipeSearchOverlay.java:445]: Bazaar Tweaks

**Review by jani270 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**